internal/clients/substrate/substrate.go

```diff
@@ -0,0 +1,74 @@
+package substrate
+
+import (
+	"fmt"
+	"math/big"
+	"strconv"
+
+	"github.com/threefoldtech/tf-kyc-verifier/internal/logger"
+
+	// use tfchain go client
+
+	tfchain "github.com/threefoldtech/tfchain/clients/tfchain-client-go"
+)
+
+type Substrate struct {
+	api    *tfchain.Substrate
+	logger logger.Logger
+}
+
+func New(config SubstrateConfig, logger logger.Logger) (*Substrate, error) {
+	mgr := tfchain.NewManager(config.GetWsProviderURL())
+	api, err := mgr.Substrate()
+	if err != nil {
+		return nil, fmt.Errorf("substrate connection error: failed to initialize Substrate client: %w", err)
+	}
+
+	c := &Substrate{
+		api:    api,
+		logger: logger,
+	}
+	return c, nil
+}
+
+func (c *Substrate) GetAccountBalance(address string) (*big.Int, error) {
+	pubkeyBytes, err := tfchain.FromAddress(address)
+	if err != nil {
+		return nil, fmt.Errorf("failed to decode ss58 address: %w", err)
+	}
+	accountID := tfchain.AccountID(pubkeyBytes)
+	balance, err := c.api.GetBalance(accountID)
+	if err != nil {
+		if err.Error() == "account not found" {
+			return big.NewInt(0), nil
+		}
+		return nil, fmt.Errorf("failed to get balance: %w", err)
+	}
+
+	return balance.Free.Int, nil
+}
+
+func (c *Substrate) GetAddressByTwinID(twinID string) (string, error) {
```

Comment:
Provide another version that works with a uint32, It's the problem of the caller to parse/convert to the correct type, you can also create another helper with string, but having uint32 one is better.